app/components/primer/alpha/action_list/heading.rb

```diff
@@ -11,21 +11,27 @@ class Heading < Primer::Component
           :filled => "ActionList-sectionDivider--filled"
         }.freeze
         SCHEME_OPTIONS = SCHEME_MAPPINGS.keys.freeze
+        HEADING_MIN = 1
+        HEADING_MAX = 6
+        HEADING_LEVELS = (HEADING_MIN..HEADING_MAX).to_a.freeze
 
         # @param list_id [String] The unique identifier of the sub list the heading belongs to. Used internally.
         # @param title [String] Sub list title.
+        # @param heading_level [Integer] Heading level. Level 2 results in an `<h2>` tag, level 3 an `<h3>` tag, etc.
         # @param subtitle [String] Optional sub list description.
         # @param scheme [Symbol] Display a background color if scheme is `filled`.
-        # @param tag [Symbol] Semantic tag for the heading.
+        # @param tag [Integer] Semantic tag for the heading.
         # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-        def initialize(list_id:, title:, tag: :h3, scheme: DEFAULT_SCHEME, subtitle: nil, **system_arguments)
-          @tag = tag
+        def initialize(list_id:, title:, heading_level: 2, scheme: DEFAULT_SCHEME, subtitle: nil, **system_arguments)
```

Comment:
There was conversation about h2 vs h3 _somewhere_ in dotcom that I would need to go digging for, but I think we landed on h3 being the default. I believe @kendallgassner was working on this..

Reply:
oof I don't remember if I was talking about a specific feature case 🤔 .  If I am looking at the [searchResult](https://github.com/search?q=search&type=repositories) navigation menu for example it makes sense for `filterBy` to be a `h2` and languages to be an `h3`. I imagine most cases would be formatted in similarly so defaulting to `h3` would make most sense to me but I do want to ensure we have the option to change the level if needed (i.e. `as='h2`)?